src/foam/nanos/auth/RetrievePassword.js

```diff
@@ -98,10 +99,18 @@ foam.CLASS({
             transient: true
           }));
           this.stack.push({ class: 'foam.u2.view.LoginView', mode_: 'SignIn' }, this);
-        })
-        .catch((err) => {
+        }).catch((err) => {
+          let id = err.data && err.data.id && err.data.id;
+          var message = this.ERROR_MSG;
+          var description;
+          if ( id ) {
+            message = foam.String.labelize(id.split('.').pop());
+            message = this.translationService.getTranslation(foam.locale, id+'.notification.message', message);
+            description = this.translationService.getTranslation(foam.locale, id+'.notification.description', err.message);
+          }
           this.ctrl.add(this.NotificationMessage.create({
-            message: err.message || this.ERROR_MSG,
+            message: message,
```

Comment:
Not part of your PR, but I'm not very keen on this.ctrl.add(). I would rather that ctrl.notify() work when you're not logged in. No need to change the PR, just mentioning.

Reply:
Created ticket to track.